### PR TITLE
Unity Plugin: compiler directive to enable builds and synchronising C# scripts with MuJoCo

### LINKED ISSUE #245: Build issue with Unity Plugin with Editor scripts
Hello, on Windows 10, Unity version 2021.2.5 running a build with MjComponents within fails due to unresolved references / type and namespace issues in existing MjComponent's Editor scripts. There are no issues due to these scripts inside the Editor, and wrapping all mujoco editor scripts (e.g. `MjGeomEditor.cs`) with the directive

```
#if UNITY_EDITOR
//Component Editor Script
#endif
```
resulted in no issues in compilation and build. Are there any specific steps that can be taken to build with the plugin without these directives? Or could this be an issue stemming from an incorrect import of the plugin on my side?

### PR DESCRIPTION
**Pull request by Balint-H**
# Description

Two small additions:

1.  The plugin's Editor scripts were not ignored during build, leading to unresolved references preventing standalone builds. The scripts that prevented the build were wrapped with the following directive:
```
#if UNITY_EDITOR
//Component Editor Script
#endif
```
Behaviour is not impacted in the editor, but builds are now successful. Fixes #245.

2.  An event was added to `MjScene` that allows C# scripts, such as custom controllers and agents to subscribe handlers that are invoked between `mj_step1` and `mj_step2`. This can improve the stability of certain actuator controllers, and can guarantee order of execution instead of relying on separate `FixedUpdate()`s. If no such handler is subscribed, then `mj_step` is executed as usual. Pointers to the simulation data for event handlers are included in a custom `EventArgs`. Related to #233.

### CODE REVIEW

**Review by erez-tom — APPROVED** [review]
Approved for review.

**Review by saran-t — APPROVED** [review]

File: unity/Editor/Components/AbsoluteValuePropertyEditor.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#if UNITY_EDITOR
```
Comment (saran-t) [inline unity/Editor/Components/AbsoluteValuePropertyEditor.cs:1]: Could you please move these to underneath the copyright header? (Here and in other files)

**Review by saran-t — CHANGES_REQUESTED** [review]

File: unity/Editor/Components/AbsoluteValuePropertyEditor.cs (hunk 1 above)
Comment (Balint-H) [reply unity/Editor/Components/AbsoluteValuePropertyEditor.cs]: Thank you for the review, I've ensured the header comes first in all files.

**Review by saran-t — APPROVED** [review]

**Comment by jespersmith** [comment]
I propose an alternative fix to the compiler directives, by configuring the .asmdef in the Editor folder correctly

https://github.com/deepmind/mujoco/pull/303

This avoids having to create a lot of #ifdef directives and keeps the code cleaner overall.

**Comment by Balint-H** [comment]
I edited this pull request to only add the events to MjScene that can be used to synchronise behaviour with C# scripts, and removed the edits to the Editor scripts, as I believe #303 handled that part better.

**Review by erez-tom — APPROVED** [review]

**Review by erez-tom — APPROVED** [review]

**Comment by Balint-H** [comment]
I will submit a separate pull request for the event synchronising functionality from a branch thats more in sync with the deepmind main.

**Pull request closed without merging**